### ISSUE TITLE
More vivid colours - especially in night/dark mode

### ISSUE THREAD

**Issue by AKDub** [organicmaps/organicmaps#688]
It could be my phone screen or my eyesight, but both themes are very muted to me, especially in night mode - it feels light I have the screen on the lowest brightness setting.

Would it be possible to have a light and dark theme that has more vivid colours?

**Comment by biodranik** [comment]
Yes, if someone with a good visual taste helps us to polish current styles :)
There is a style editor in the code, you may try to play with it too.

**Comment by maciboy** [comment]
Please, do not try purple… Others have already tried :P

**Comment by Raphael0** [comment]
If there is a revamp of the dark map theme, please keep the absolute black (#000000) background.

One of the nicest things if you have an OLED device is to follow lines appearing on an otherwise completely dark screen e.g. when cycling at night :) Some other apps use a dark grey background which looks very different in such a setting.

**Comment by biodranik** [comment]
From user's feedback to improve the contrast:

1 | 2
---|---
![image](https://user-images.githubusercontent.com/170263/156162375-64c8d509-b633-45a6-bbf8-1926935305ff.png) | ![image](https://user-images.githubusercontent.com/170263/156162394-32f26b4a-8eaf-4fbc-9149-e277df59779d.png)
![image](https://user-images.githubusercontent.com/170263/156162423-8232d6b3-4a4f-41ba-a409-c18d65aee489.png) | ![image](https://user-images.githubusercontent.com/170263/156162452-4f0aa6b8-9b73-458e-924e-f29a283035ee.png)
![image](https://user-images.githubusercontent.com/170263/156162480-2d01a790-ffe2-41c5-a3d0-97d13e5bd1ee.png) | ![image](https://user-images.githubusercontent.com/170263/156162497-4df78455-3d22-425e-b320-8b7fe57c4a5e.png)

**Comment by biodranik** [comment]
Current styles implementation is complicated. It may take a lot of time to make it easily editable by the community.
See also #1577

**Comment by pastk** [comment]
There seems to be 2 use cases:
- a night-specific mode with low contrast - e.g. important during night navigation not to dazzle a driver
- a dark all-around mode with high contrast, as its supposed to be used in sunlight as well - its more a matter of visual preference (and battery saving for e.g. oled screens)

I'm not sure its possible to serve both use cases well by one map style. The current OM night style is of the first type, its unusable at day time.

I wonder if just reducing screen's brightness would low down the contrast enough to use an otherwise high-contrast dark mode at night comfortably? How do other map apps handle it?

**Comment by AKDub** [comment]
I'm not sure a low contrast dark mode is needed is it? Surely if the bulk of the light background is changed to dark, then the driver is less likely to be dazzled, and really isn't it up to the driver to adjust the brightness of the screen? For me that is the main issue - when I turn down the screen brightness at night, I lose the ability to clearly see the information.

Excuse my awful drawings, but this is how I see it. I have used colours off the top of my head that are generally used in UK maps (you get the idea)

Dark blue - Motorways
Dark green - smaller roads
Light Blue - Water/rivers
Light Green - Fields/Trees
Grey - Buildings/Urban areas
Purple - The route to follow

As you can see, in my mind the details don't change colour, just the background does. I know it's more complicated than this, but surely this is what most people mean by a dark mode?

<img src="https://user-images.githubusercontent.com/43154960/165596246-b34565ff-0377-414a-a229-d2b4caf094eb.png" width=50% height=50%>

<img src="https://user-images.githubusercontent.com/43154960/165596253-2d81d522-74a9-4e2e-8a3c-90eaba66a5bd.png" width=50% height=50%>

**Comment by biodranik** [comment]
These colors are too... colorful. Which apps are using it in the UK?

**Comment by AKDub** [comment]
Again, excuse my poor atempt to draw this, I should have probably just found images. The colours I chose dont really matter, they were just off the top of my head as the colours used in UK road maps. Any app that offers country specific themes would have them. 

The colours are too garish as I've drawn them, but the main point I was making was that dark mode does not neccesarliy need to be slightly varying degrees of black/grey (or white/grey for that matter in light mode).

**Comment by julianfairfax** [comment]
> Yes, if someone with a good visual taste helps us to polish current styles :)
> There is a style editor in the code, you may try to play with it too.

I'm not going to come out here and say that I have "good visual taste", but, in my humble opinion, the green colour on the app icon is too dark. 

I think it'd be nicer if it resembled more the green for fields in the app, or somewhere in between.

![Screenshot_20220705-025037_1.png](https://user-images.githubusercontent.com/43753131/177229590-4d318320-c298-4478-ae21-ce03dd25c7e9.png)

**Comment by chereskata** [comment]
+1 for improving on the night mode to be brighter

**Comment by d4f5409d** [comment]
In my opinion we should rework light and dark mode to look more like Mapbox and then have a high contrast mode, as a setting if someone needs it.

**Comment by biodranik** [comment]
The current dark style targets drivers at night. This is a very important use case where a dark style with low contrast is needed to avoid pain in the eyes.

A "dark mode for daylight" may be created for those who like dark maps during the day. But it is a very different use-case.

**Comment by d4f5409d** [comment]
> The current dark style targets drivers at night. This is a very important use case where a dark style with low contrast is needed to avoid pain in the eyes.
> 
> A "dark mode for daylight" may be created for those who like dark maps during the day. But it is a very different use-case.

No, we're not talking about a daylight dark mode. We're just talking about improving eyesight in dark mode.

**Comment by biodranik** [comment]
There is also an idea to optimize the theme for OLED (AMOLED) screens by making greyish colors even darker. Considering that iPhones 13+ have OLED screens, that can be beneficial for some users battery-wise, and have a better experience at night.

So we need to support two different modes: dark for the night (LCD + OLED?) and dark for sunlight (a more contrast one).

**Comment by d4f5409d** [comment]
> There is also an idea to optimize the theme for OLED (AMOLED) screens by making greyish colors even darker. Considering that iPhones 13+ have OLED screens, that can be beneficial for some users battery-wise, and have a better experience at night.
> 
> So we need to support two different modes: dark for the night (LCD + OLED?) and dark for sunlight (a more contrast one).

Can just we get something like Maps.me does? They really nailed the colors.
![Screenshot_2023-09-14-10-04-29-40_aad5a51bd03a71d801917edbc0ab6f95.jpg](https://github.com/organicmaps/organicmaps/assets/76624594/ba6f5681-9bcc-4dc8-a17c-5d42f37a04f6)

![Screenshot_2023-09-14-10-04-49-49_aad5a51bd03a71d801917edbc0ab6f95.jpg](https://github.com/organicmaps/organicmaps/assets/76624594/6e12ddaa-eacd-43a5-aaac-1dfda9d4bf8f)

**Comment by chereskata** [comment]
> So we need to support two different modes: dark for the night (LCD + 
OLED?) and dark for sunlight (a more contrast one).

OLED / LCD Mode could be by toggle and the sunlight mode from sensors or 
from quick access toggle. Unfortunately i do not know if there exists an 
API to easily query such information regarding sun brightness hitting 
the screen.

> Can just we get something like Maps.me does? They really nailed the colors.

i like it.

**Comment by nimxaa** [comment]
True black causes smearing on OLED screens so it would be nice to have dark theme that doesn't use it.

**Comment by j-lakeman** [comment]
I'd very much like to see AMOLED or “true black” theme for two main reasons:

1. potential to save a lot of battery
2. reducing eye strain when driving at night

**Comment by dvdmrtnz** [comment]
I will take a look at improving the colors in nightmode. I have prepared this svg with all the colors that are currently in use in light and dark mode, in case someone else wants to take a crack at redesigning the map colors!

Updated 2025-12-29:

![](https://github.com/user-attachments/assets/894168f0-fc44-4af8-a76e-d8a667a71ddd)

**Comment by Jean-BaptisteC** [comment]
Nice work, maybe add on Wiki?

**Comment by biodranik** [comment]
The current dark style was developed to drive the car at night. Does it make sense to make the low contrast dark vehicle-only style when car navigation mode is active, and use a better contrast style suitable also for the daylight in all other map modes, to avoid switching between two dark mode styles?

Now we have:
1. Low contrast normal style (suitable for dark car environment)
2. Similar low-contrast vehicle style (active in navigation only)

Proposal:
1. High contrast normal style
2. Low contrast AMOLED vehicle style (active in navigation mode only).

More complex alternate approach:
1. High contrast normal style.
2. High contrast vehicle/driving style
3. Low contrast normal style
4. Low contrast vehicle/driving style

**Comment by dvdmrtnz** [comment]
@biodranik Yes, I think that’s a good approach. Keep the low contrast style in vehicle night mode and add some more contrast to the regular night mode.

**Comment by dvdmrtnz** [comment]
> I worry you may reduce the contrast too much, especially with CarPlay and Android Auto, since people need to be able to clearly and easily see the map. As long as the background is dark, the screen brightness settings are much more important than low contrast.

AFAIK contrast is already low. What needs to be done is increase contrast in regular night mode while keeping low contrast in vehicle night mode as it is

**Comment by Korb** [comment]
I agree that the current implementation of the dark color theme is questionable. It is, at a minimum, not universal, and should be changed or supplemented with another, more contrasting theme. As a smartphone user with an OLED screen, I prefer to use only the dark theme, regardless of the time of day. However, Organic Maps is one of the few apps that I am forced to use a light color theme. The thing is, even with maximum screen brightness in cloudy weather in the open air, I only see the current location cursor, metro lines and round interface buttons. The card itself is almost indiscernible.

Here is the current view of the application's main screen:
![Screenshot_20240602_181213](https://github.com/organicmaps/organicmaps/assets/3509797/a1c50a18-834f-486f-b258-2439be1cd90c)

And here is what I think will be suitable for use not only in the dark, but also during the day, unlike the current dark theme:
![Screenshot_20240602_181213](https://github.com/organicmaps/organicmaps/assets/3509797/b0583d61-ae68-41e8-80ee-29016a41bddb)

**Comment by XenonSup** [comment]
> ... I only see the current location cursor, metro lines and round interface buttons. The card itself is almost indiscernible.
> 
> Here is the current view of the application's main screen: 

@Korb you're probably already aware of this, but just so it's out there: Turning no the subway layer lowers the contrast of the map in both light and dark modes.
I personally find that it's too dramatic and makes the rest of the map borderline unusable, but I do see the reasoning behind doing it. Perhaps it just needs to be toned down a little, or disabled altogether in dark mode since it's already low-contrast and the colorful subway lines will pop regardless.


I also agree with (my interpretation of) @rtsisyk 's "simple" proposal
- Add high-contrast dark theme as default everywhere except car navigation
- Polish (to cover OLED screens with pure blacks) existing low-contrast dark theme for in-vehicle navigation

And to address different users' use-cases, add a 3-way option "Dark Theme":
1. Use low contrast dark theme in-car only (default)
2. Use low contrast dark theme always
3. Use high contrast dark theme always


Should option 1. cover active navigation only, or "just drive" mode as well?

**Comment by pastk** [comment]
@XenonSup @Korb you might be interested to check the following idea that removes the half-opaque subway underlay:
- #8431

**Comment by Misalf-git** [comment]
> As a smartphone user with an OLED screen, I prefer to use only the dark theme, regardless of the time of day. However, Organic Maps is one of the few apps that I am forced to use a light color theme.

Absolutely, except that I don't even need an OLED display to prefer dark themes.

**Comment by Korb** [comment]
> 1. Use low contrast dark theme in-car only (default)

@XenonSup, do you mean that after successfully creating a route for the car and pressing the Start button, the dark high-contrast theme will be switched to a dark, low-contrast theme?

> Should option 1. cover active navigation only, or "just drive" mode as well?

I do not understand what do you mean. Other navigation apps have a car mode without navigation. But I didn't know that it was also in Organic Maps. Perhaps its activation should be placed in a more obvious place. But this is a question for a separate issue.

> you might be interested to check the following idea that removes the half-opaque subway underlay Remove subways bg, add line casing

@pastk, even after removing the translucent backing for the subway lines, there is nothing like white on the map (the dark theme is on). `#8e8d92`, `#9ca5ac`, `#a4a9ac`... - all this is very far from white, the color of application controls. Although, I admit, the map became a little more legible when outdoors.

**Comment by DadiBit** [comment]
A quick addition: both my iPhone and Xiaomi have a monochromatic filter (set through accessibility), which pretty much kills vivid colors. This pretty much relies only on the contrast in terms of visual elements, however the color schemes (dark one in particular) doesn't rely on the contrast enough to let the app be usable.

I'm not an expert in terms of UI/colors and accessibility, however I would like to ask if it would be possible to increase the contrast, rather than working on the saturation (since it's non-existent in monochromatic views)

**Comment by oleg-rswll** [comment]
**Google Play**
1/8/25

> Navigator mode (at night) - I would like more contrast roads and POI. You can't see everything except the route line.

**Comment by ghost** [comment]
Are there any updates on this issue? actually when i am in my dark room at night i still barely see the map in night mode and actually this is (almost) the only thing that keeps me away from using it because i cannot see anything in dark mode, i cant imagine see it while driving at all, i have an IPS panel, its not the highest contrast but its pretty good and still cannot see anything, i would gladly help to resolve this issue, because theres no other alternatives to google maps so this app should become the one because it is pretty close

**Comment by antoniocostaamaral** [comment]
Agreed. A separate high-contrast dark mode is necessary - dark mode's usability is otherwise limited to narrow low-light conditions.

The implementation of the new mode could start simple: just boost the luminosity of existing colors across the board, then fine-tune as updates roll out.

An alternative: a button to switch to, or temporarily peek into (hold to peek, release to revert), the opposite mode.

**Comment by 1HitKilobyte** [comment]
just want to throw my feedback on the pile. the dark mode is absurdly dark and a tiny fraction of the brightness of all my other dark mode apps and the contrast is not accessible.

I checked the latest version as of posting and the contrast is 1.44:1 which is less than half the bare minimum suggestion by WCAG for standard text and healthy vision (3:1 but for better accessibility they suggest it should be 4.5:1 or 7:1).

I have laser vision correction and excellent vision but this app is unusable to me. I would like to see a dark mode that has at least 4.5:1 contrast for the roads/etc and labels and allows vivid and distinct colors.

**Comment by biodranik** [comment]
Please don't forget that the dark mode in Organic Maps initially was developed for car drivers at night, not for the daily use under a bright sun.

**Comment by 1HitKilobyte** [comment]
I appreciate OM considering the navigation experience for drivers first and indeed that mode is accessible and easy to see with better contrast and slightly better colors. the route preview also has these corrections.

I just hope eventually that other use cases will be considered. I would love to switch if it becomes accessible.

**Comment by antoniocostaamaral** [comment]
Nowadays dark mode is the system default for most people, and plenty of apps (email, chat, social media, and the OS itself) handle it quite well even in bright daylight.

With all due respect and appreciation for the hard work of the Organic Maps support and development team, I'd say that for a maps/navigation app - which is often used outdoors - the daylight experience in dark mode shouldn't be described as "unusable".

I realize remapping all the colors would be a lot of work. As a simpler solution, could you add a high-contrast toggle for the dark theme?

**Comment by antoniocostaamaral** [comment]
I went for a long hike, with bright sunlight, and what a miserable experience i had with dark mode on organic maps. 

Even shielding the screen from direct sunlight, the contrast is so dim that it's almost unusable. you cannot figure out the paths ahead, you risk getting lost. 

In light mode the contrast is not that much better, and my loaded kmz, thought for dark mode, show with poor colors. 

I love the app, i realise it's a lot of work, but come on, for a map app, this is is truly a design flaw for the ages.
